pkg/cmd/roachtest/roachtestutil/mixedversion/planner_test.go

```diff
@@ -310,35 +312,39 @@ func Test_maxNumPlanSteps(t *testing.T) {
 	// N.B. the upper bound is very conservative; largest "basic upgrade" plan is well below it.
 	require.LessOrEqual(t, plan.length, 100)
 
-	mvt = newBasicUpgradeTest(MaxNumPlanSteps(15))
-	plan, err = mvt.plan()
-	require.NoError(t, err)
-	require.LessOrEqual(t, plan.length, 15)
+	mvt = newBasicUpgradeTest(MaxNumPlanSteps(13))
+	require.Eventually(t, func() bool {
+		plan, err = mvt.plan()
+		if err != nil {
+			return false
+		}
+		fmt.Println(plan.PrettyPrint())
+		require.LessOrEqual(t, plan.length, 13)
+		return true
+	}, 5*time.Second, 100*time.Millisecond)
 
-	// There is in fact no "basic upgrade" test plan with fewer than 15 steps.
+	// There is in fact no "basic upgrade" test plan with fewer than 13 steps.
```

Comment:
It looks like 13 is the actual smallest plan. The original plan commented ran the `mixed-version 2` hook twice and used fixtures (70%) chance.